pkg/ddc/juicefs/data_migrate.go

```diff
@@ -215,6 +220,46 @@ func (j *JuiceFSEngine) generateDataMigrateValueFile(r cruntime.ReconcileRequest
 	return valueFile.Name(), nil
 }
 
+func addWorkerPodAntiAffinity(dataMigrateInfo *cdatamigrate.DataMigrateInfo, dataMigrate *datav1alpha1.DataMigrate) {
+	releaseName := utils.GetDataMigrateReleaseName(dataMigrate.Name)
+
+	podAffinityTerm := corev1.WeightedPodAffinityTerm{
+		Weight: 100,
+		PodAffinityTerm: corev1.PodAffinityTerm{
+			LabelSelector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					dataoperation.OperationLabel: fmt.Sprintf("migrate-%s-%s", dataMigrate.Namespace, releaseName),
+				},
+			},
+			TopologyKey: "kubernetes.io/hostname",
+		},
+	}
+
+	// Affinity is nil
+	if dataMigrateInfo.Affinity == nil {
+		dataMigrateInfo.Affinity = &corev1.Affinity{
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+					podAffinityTerm,
+				},
+			},
+		}
+		return
+	}
+	// Affinity not nil, PodAntiAffinity is nil
+	if dataMigrateInfo.Affinity.PodAntiAffinity == nil {
+		dataMigrateInfo.Affinity.PodAntiAffinity = &corev1.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+				podAffinityTerm,
+			},
+		}
+		return
+	}
+	// PodAntiAffinity not nil
+	dataMigrateInfo.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution =
+		append(dataMigrateInfo.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution, podAffinityTerm)
```

Comment:
`append` can be applied to nil slice, so I think we can simplify this part with:
```
if dataMigrateInfo.Affinity.PodAntiAffinity == nil {
    dataMigrateInfo.Affinity.PodAntiAffinity = &corev1.PodAntiAffinity{}
}

dataMigrateInfo.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution =
		append(dataMigrateInfo.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution, podAffinityTerm)
```

Reply:
fixed.